### DISCUSSION TITLE
Parse Solidity Enum using wagmi/cli

### DISCUSSION THREAD

**Discussion by 0xMySt1c** [wagmi-dev/abitype#121] [Ideas]
I wasn't sure originally if this was better suited in ABITypes, Viem, or wagmi, but after some thought I think parsing the shape of a solidity enum into typescript is the perfect job for wagmi/cli. Since enums are mapped to uint8 and don't actually show up in ABIs, but do show up in artifacts folder.

**Comment by tmm** [comment]
Interesting idea! Moving this over to ABIType for discussion. Have a rough example of what it would look like or how you would expect it to function (inputs/outputs)?

**Reply by 0xMySt1c** [reply]
I actually meant to put this in the wagmi repo!

**Reply by 0xMySt1c** [reply]
I will move the discussion to wagmi.

**Comment by 0xMySt1c** [comment]
https://github.com/wagmi-dev/wagmi/discussions/2180

**Comment by Raiden1411** [comment]
The idea here is to use the generated artifacts from foundry/hardhat into typescript enums? Or it would also be for "human-readable" enum signatures?
